libr/io/section.c

```diff
@@ -57,8 +57,8 @@ static RIOSection *findMatching (RIO *io, ut64 paddr, ut64 vaddr, ut64 size, ut6
 R_API RIOSection *r_io_section_add(RIO *io, ut64 offset, ut64 vaddr, ut64 size, ut64 vsize, int rwx, const char *name, ut32 bin_id, int fd) {
 	int update = 0;
 	RIOSection *s;
-	if (size == 0 || size > 0xf0000000) {
-		if (size > 0 && size != UT64_MAX && size != UT32_MAX)
+	if (size == 0) {
+		if (size > 0 && size != UT64_MAX && size != UT32_MAX) //hacky things which might give bad output in case size == UT32_MAX for 64bit elf.
```

Comment:
```C
	if (size == 0) {
		if (size > 0 && size != UT64_MAX && size != UT32_MAX) //hacky things which might give bad output in case size == UT32_MAX for 64bit elf.
			eprintf ("Invalid size (0x%08" PFMT64x
				 ") for section '%s' at 0x%08" PFMT64x "\n",
				 size, name, vaddr);
		return NULL;
	}

```

you only will reach that statement when `size == 0` and then `size > 0` will never be true unless i am missing something. I think it should be

```C
	if (!size && size == UT64_MAX && size == UT32_MAX) //hacky things which might give bad output in case size == UT32_MAX for 64bit elf.
		eprintf ("Invalid size (0x%08" PFMT64x
			 ") for section '%s' at 0x%08" PFMT64x "\n",
			 size, name, vaddr);
	return NULL;
```